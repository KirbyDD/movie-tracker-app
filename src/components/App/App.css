* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  text-align: center;
}

.background-image {
  width:100%;
  height: 100%;
  position: fixed;
  left: 0px;
  opacity: 0.9;
  top: 0px;
  z-index: -1;
}

.App-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-image: linear-gradient(#141414, #2c2b2b);
  height: 50px;
  font-size: 16px;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
}



.App-header-login {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 50px;
  font-size: 16px;
  position: fixed;
  top: 0;
  left: 0;
  width: 50vw;
}



.login-name,
.favorites-page {
  color: #fff;
  text-decoration: none;
  padding-right: 32px;
}

.login-name:hover {
  color: #01D277
}

.logo {
  color: #01D277;
  padding-left: 20px;
}

.logo-tracker {
  color: #fff;
}

.logo-signup {
  font-size: 32px;
  font-weight: bold;
  text-decoration: none;
}

.App-body {
  margin-top: 50px;
  width: 100vw;
}

.footer {
  background: black;
  width: 100vw;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
}

.footer-text {
  color: rgba(255, 255, 255, 0.466);
}

.logout-container {
  border-radius: 5px;
  background-image: linear-gradient(#141414, #2c2b2b);
  width: 200px;
  height: 100px;
  position: absolute;
  right: 25px;
  top: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.logout-btn {
  width: 70%;
  height: 20px;
  border-radius: 3px;
  background: rgba(1, 210, 120, 0.753);
  cursor: pointer;
  outline: none;
}

.logout-btn:hover {
background: #01D277;
}

.logout-arrow {
  background-image: linear-gradient(#141414, #2c2b2b);

  width: 20px;
  height: 20px;
  position: absolute;
  top: -5px;
  right: 9px;
  transform: rotate(45deg);
  z-index: -1;
}