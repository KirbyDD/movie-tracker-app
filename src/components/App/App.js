import React from 'react';
import MovieContainer from '../MovieContainer/MovieContainer';
import User from '../User/User'
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>MOVIE TRACKER</h1>
      </header>
      <User />
      <MovieContainer />
      <footer>
        <p>Powered by TMDB</p>
      </footer>
    </div>
  );
}

export default App;
