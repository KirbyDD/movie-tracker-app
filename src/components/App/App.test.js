import React from 'react';

import { shallow } from 'enzyme';
import {App } from './App';

describe('App', ()=> {

  it('should match the snapshot', ()=> {
    let currentUser ={loggedIn: false, userDetail: {name: 'someone'}}
    let wrapper = shallow(<App currentUser={currentUser}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if user is logged in', ()=> {
    let currentUser ={loggedIn: true, userDetail: {name: 'someone'}}
    let wrapper = shallow(<App currentUser={currentUser}/>);
    expect(wrapper).toMatchSnapshot();
  })

  });
