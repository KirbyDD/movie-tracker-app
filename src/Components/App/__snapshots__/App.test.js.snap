// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movie/:movie_id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;

exports[`App should match the snapshot if user is logged in 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movie/:movie_id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;
